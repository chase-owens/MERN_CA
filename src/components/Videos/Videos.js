import React from 'react';
import { theme } from '../../styles/theme';

import withWidth from '@material-ui/core/withWidth';
import { unstable_useMediaQuery as useMediaQuery } from '@material-ui/core/useMediaQuery';
import { withStyles } from '@material-ui/core/styles';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

import videos from './videos.config';
import VideoCard from '../VideoCard/VideoCard';
import { isMobile } from 'react-device-detect';

const styles = theme => ({
  videoCard: {
    [theme.breakpoints.down(740)]: {
      width: '100%'
    },
    [theme.breakpoints.up(740)]: {
      width: '50%'
    },
    [theme.breakpoints.up('md')]: {
      width: '33.3%'
    }
  }
});

const Videos = ({ classes }) => {
  return isMobile ? (
    <div style={{}}>
      <Typography
        paragraph
        variant='caption'
        style={{ marginLeft: 100, fontSize: '2.1em', marginBottom: 45 }}
      >
        <span style={{ fontWeight: 'bold' }}>CA</span> / Videos
      </Typography>
      <Typography style={{ fontSize: '3em' }} paragraph variant='headline'>
        Videos
      </Typography>
      <Grid
        container
        spacing={0}
        justify='space-between'
        style={{
          width: '100vw'
        }}
      >
        {videos.map(video => (
          <Grid
            item
            key={video.title}
            style={{ width: '90vw', margin: 'auto' }}
          >
            <VideoCard video={video} />
          </Grid>
        ))}
      </Grid>
    </div>
  ) : (
    <div style={{ padding: 40, margin: 'auto', overflow: 'hidden' }}>
      <Typography paragraph variant='caption'>
        <span style={{ fontWeight: 'bold' }}>CA</span> / Videos
      </Typography>
      <Typography style={{ fontSize: '2em' }} paragraph variant='headline'>
        Videos
      </Typography>
      <Grid container spacing={40} justify='flex-start'>
        {videos.map(video => (
          <Grid item key={video.title} className={classes.videoCard}>
            <VideoCard video={video} />
          </Grid>
        ))}
      </Grid>
    </div>
  );
};

export default withStyles(styles)(withWidth({ withTheme: true })(Videos));
