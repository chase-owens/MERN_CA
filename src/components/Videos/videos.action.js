export const toggleMovie = video => ({
  type: TOGGLE_MOVIE,
  video
});

export const TOGGLE_MOVIE = 'TOGGLE_MOVIE';
