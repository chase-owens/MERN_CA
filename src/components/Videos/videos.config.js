const videos = [
  {
    videoPath: `https://youtu.be/tJLZ82k_HE0`,
    title: `Interaction Guidelines Nanook`,
    description: `An example of how to carryout the Interaction Guidelines`,
    imagePath: `Nanook`,
    embedPath: `https://www.youtube.com/embed/tJLZ82k_HE0`
  },
  {
    videoPath: `https://youtu.be/A0WBNze3oFY`,
    title: `Interaction Guidelines Beagle`,
    description: `An example of how to carryout the Interaction Guidelines`,
    imagePath: `Blackie`,
    embedPath: `https://www.youtube.com/embed/A0WBNze3oFY`
  },
  {
    videoPath: `https://youtu.be/D6mma4s0pEU`,
    title: `Affection Loop at Home`,
    description: `Teaching Jerry how to remain calm while his Mom is exercising, talking on the phone, and bringing in the groceries`,
    imagePath: `Jerry`,
    embedPath: `https://www.youtube.com/embed/D6mma4s0pEU`
  },
  {
    videoPath: `https://youtu.be/G0ZCKGtUfbU`,
    title: `Affection Loop in Shelters`,
    description: `Teaching shelter dogs to sit patiently while meeting and greeting potential adopters`,
    imagePath: `Shelter`,
    embedPath: `https://www.youtube.com/embed/G0ZCKGtUfbU`
  },
  {
    videoPath: `https://youtu.be/HbWi8LVAKxU`,
    title: `Ranatta is too Turnt`,
    description: `Watch Ranatta transform from an energetic puppy to a calm companion`,
    imagePath: `Ranatta`,
    embedPath: `https://www.youtube.com/embed/HbWi8LVAKxU`
  },
  {
    videoPath: `https://youtu.be/59aVwQRB_rM`,
    title: `Preventing Separation Anxiety`,
    description: `Practicing Patience`,
    imagePath: `Jax`,
    embedPath: `https://www.youtube.com/embed/59aVwQRB_rM`
  },
  {
    videoPath: `https://youtu.be/lCvlDLLpges`,
    title: `Rocky is Really Ready`,
    description: `Rocky learns to be more reserved before going on walks helping them get off on the right foot`,
    imagePath: `Rocky`,
    embedPath: `https://www.youtube.com/embed/lCvlDLLpges`
  },
  {
    videoPath: `https://youtu.be/WKtZuMQ4cJU`,
    title: `Stay Peepers`,
    description: `Adding a 'Stay' cue during the Affection Loop`,
    imagePath: `Peeper`,
    embedPath: `https://www.youtube.com/embed/WKtZuMQ4cJU`
  }
];

export default videos;
