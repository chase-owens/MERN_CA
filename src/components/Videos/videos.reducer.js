import { TOGGLE_MOVIE } from "./videos.action";
import { defaultState } from "../../server/defaultState";

export const videosReducer = (state = defaultState.video, action) => {
  switch (action.type) {
    case TOGGLE_MOVIE:
      return { video: action.video };
    default:
      return state;
  }
};
