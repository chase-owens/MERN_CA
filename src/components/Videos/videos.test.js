import React from 'react';
import Videos from './Videos';
import { shallow } from 'enzyme';

describe('Videos', () => {
  it('renders without crashing', () => {
    const videos = shallow(<Videos />);
    expect(videos).toMatchSnapshot();
  });
});
