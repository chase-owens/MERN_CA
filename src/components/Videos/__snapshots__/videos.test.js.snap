// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos renders without crashing 1`] = `
<WithTheme(WithWidth(Videos))
  classes={
    Object {
      "videoCard": "WithTheme-WithWidth-Videos---videoCard-1",
    }
  }
/>
`;
